ufc/bandit-tests/test-case-car-bandit.json

```diff
@@ -0,0 +1,50 @@
+{
+    "flag": "car_bandit_flag",
+    "defaultValue": "default",
+    "subjects": [
+      {
+        "subjectKey": "anastasia",
+        "subjectAttributes": {},
+        "actions": [
+            {
+                "actionKey": "dodge",
+                "numericAttributes": {},
+                "categoricalAttributes": {}
+            },
+            {
+                "actionKey": "mercedes",
+                "numericAttributes": {},
+                "categoricalAttributes": {}
+            },
+            {
+              "actionKey": "toyota",
+              "numericAttributes": {"speed":  1000},
+              "categoricalAttributes": {}
+            }
+        ],
+        "assignment": {"variation": "car_bandit", "action": "dodge"}
+      },
+      {
+        "subjectKey": "bob",
+        "subjectAttributes": {},
+        "actions": [
+            {
+              "actionKey": "toyota",
+              "numericAttributes": {"speed":  1},
+              "categoricalAttributes": {}
+            },
+            {
+              "actionKey": "honda",
+              "numericAttributes": {},
+              "categoricalAttributes": {}
+            },
+            {
+              "actionKey": "ford",
+              "numericAttributes": {},
+              "categoricalAttributes": {}
+            }
+        ],
+        "assignment": {"variation": "car_bandit", "action": "ford"}
```

Comment:
The above actions shuffle to: Toyota, Honda, Ford

If the default action score is ignored and the unknown (non-Toyota) actions get scored `0`, Toyota takes up too many shards and that is what is assigned.

With the default score at play, the two unknown actions are top-scoring and get most of the shards. However, because of the top scoring action weight of 1 - all other action weights, the top scoring action gets the most.

When not tie-broken, implementations like JavaScript that iterate over them in order would declare Honda the top-scoring action, and it would end up getting assigned. However, when breaking ties by action name, Ford takes the prize and gets assigned.

Thus, for this test to pass with Ford being assigned to Bob, the default action scores need to be respected, and top-scoring actions need to be tie-broken by name.

Reply:
hot dang, this is great test crafting; thanks for the detailed comment.

1) I thought the decision in the end was to not worry about the precision of tiebreaking the top score by name, but it really is a simple change required in the other SDKs to support this, and it eliminates the ambiguity (while arbitrary, it will at least be consistent)

Would adding a `remarks` field to the `assignment` object in the JSON break parsing in the SDK test suites? It would be really helpful to have at least some of your detailed explanations of the test case captured in the file.

Reply:
I love the `remarks` idea, and I don't think it will break anything. I was just lamenting how JSON doesn't have comment support.

Good spot on `banditReferences` 🤦 

You're right about the initial decision to not worry about tie-breaking; I had forgotten that. I think we do update the SDKs it was really easy to do for JS and may matter in situations of default score (e.g,. a cold-started bandit where everything starts with the default score)

Reply:
Ok I added as a `comment` field to each test case subject. Unfortunately no easy way to do multi-line in JSON so they are looooong lines of text. But I used the pyramid principle!